sklearn/pipeline.py

```diff
@@ -1644,23 +1642,42 @@ def fit(self, X, y=None, **fit_params):
             Targets for supervised learning.
 
         **fit_params : dict, default=None
-            Parameters to pass to the fit method of the estimator.
+            - If `enable_metadata_routing=False` (default):
+                Parameters directly passed to the `fit` methods of the
+                transformers.
+
+            - If `enable_metadata_routing=True`:
+                Parameters safely routed to the `fit` methods of the
+                transformers. See :ref:`Metadata Routing User Guide
+                <metadata_routing>` for more details.
+
+            .. versionchanged:: 1.5
+                `**fit_params` can be routed via metadata routing API
 
         Returns
         -------
         self : object
             FeatureUnion class instance.
         """
-        _raise_for_unsupported_routing(self, "fit", **fit_params)
-        transformers = self._parallel_func(X, y, fit_params, _fit_one)
+        if _routing_enabled():
+            routed_params = process_routing(self, "fit", **fit_params)
+        else:
+            # TODO(SLEP6): remove when metadata routing cannot be disabled.
+            routed_params = Bunch()
+            for name, _ in self.transformer_list:
+                routed_params[name] = Bunch(fit={})
+                routed_params[name].fit = fit_params
+
+        transformers = self._parallel_func(X, y, _fit_one, routed_params)
+
         if not transformers:
             # All transformers are None
             return self
 
         self._update_transformer_list(transformers)
         return self
 
-    def fit_transform(self, X, y=None, **fit_params):
+    def fit_transform(self, X, y=None, **params):
```

Comment:
We renamed it, because with metadata routing, the params can get routed from `fit_transform` to `transform`, if the transformer used implements `fit` and `transform` separately.

Without metadata routing it's still only routed to fit, to preserve the old behaviour.

So, to satisfy both ways to pass metadata, `params` is probably the best name, I thinks.

See, [this comment]( https://github.com/scikit-learn/scikit-learn/pull/28205#discussion_r1487743644) and [that one](https://github.com/scikit-learn/scikit-learn/pull/28205#discussion_r1497883527).

Reply:
OK fine with the name then. But subsequent questions:

- it means that we should implement the routing in `transform` as well
- do we have a plan to modify `TransformerMixin` such that `fit_transform` behave exactly the same than `fit` + `transform`?

Reply:
Not sure why we would modify `TransformerMixin`. In scikit-learn we pretty much never have metadata in `transform`. But if a third party estimator has a `.transform` that accepts metadata, then we would route it correctly, don't we?

Reply:
What I understand from this is that it depends on the transformer and what its intention is. I will keeps this in mind when working in transformers.